hare4/hare.go

```diff
@@ -550,8 +550,11 @@ func (h *Hare) Handler(ctx context.Context, peer p2p.Peer, buf []byte) error {
 			zap.Uint32("lid", msg.Layer.Uint32()),
 			zap.Stringer("sender", equivocation.Messages[0].SmesherID))
 		proof := equivocation.ToMalfeasanceProof()
-		if err := identities.SetMalicious(
-			h.db, equivocation.Messages[0].SmesherID, codec.MustEncode(proof), time.Now()); err != nil {
+		err := h.db.WithTx(context.Background(), func(tx sql.Transaction) error {
+			return identities.SetMalicious(
+				h.db, equivocation.Messages[0].SmesherID, codec.MustEncode(proof), time.Now())
+		})
```

Comment:
As @poszu already remarked for `hare3`: does this need to be a `WithTx`? If yes, `identities.SetMalicious` should be passed `tx` instead of `h.db` or otherwise it deadlocks on write I believe.

Reply:
yes it does, i answered for the reason in the original comment, it is also likely the reason why all tests failed